src/routes/open/container/Open.tsx

```diff
@@ -78,21 +79,44 @@ const getSafePropsValuesFromQueryParams = (queryParams: SafeCreationQueryParams)
 }
 
 export const createSafe = async (values: CreateSafeValues, userAccount: string): Promise<TransactionReceipt> => {
-  const confirmations = getThresholdFrom(values)
-  const ownerAddresses = getAccountsFrom(values)
-  const safeCreationSalt = getSafeCreationSaltFrom(values)
-  const deploymentTx = getSafeDeploymentTransaction(ownerAddresses, confirmations, safeCreationSalt)
-
-  const receipt = await deploymentTx
-    .send({
-      from: userAccount,
-      gas: values?.gasLimit,
-    })
-    .once('transactionHash', (txHash) => {
-      saveToStorage(SAFE_PENDING_CREATION_STORAGE_KEY, { txHash, ...values })
-    })
-
-  return receipt
+  return new Promise((resolve, reject) => {
+    const confirmations = getThresholdFrom(values)
+    const ownerAddresses = getAccountsFrom(values)
+    const safeCreationSalt = getSafeCreationSaltFrom(values)
+    const deploymentTx = getSafeDeploymentTransaction(ownerAddresses, confirmations, safeCreationSalt)
+    let receiptResult
+    const waitForTxReceipt = (resolve, reject, tries = 0) => {
+      setTimeout(() => {
+        if (receiptResult) {
+          resolve(receiptResult)
+        } else {
+          if (tries > 1200) {
+            reject(new Error('Transaction was not mined within 10 minutes'))
+          } else {
+            waitForTxReceipt(resolve, reject, tries + 1)
+          }
+        }
+      }, 500)
+    }
```

Comment:
Looks good! It would be nice to extract this into a utility so that this setTimeout loop code doesn't clutter the safe creation function.

Signature suggestion: `waitWithTimeout(comparator, onSuccess, onError, delay = 500, maxTimes = 1200)`.

Reply:
I solved it in a simpler way, you can see the change in the last commit